pkg/api/write.go

```diff
@@ -104,3 +113,44 @@ func getCounterValue(counter prometheus.Counter) float64 {
 	}
 	return dtoMetric.GetCounter().GetValue()
 }
+
+func validateWriteHeaders(w http.ResponseWriter, r *http.Request) bool {
+	// validate headers from https://github.com/prometheus/prometheus/blob/2bd077ed9724548b6a631b6ddba48928704b5c34/storage/remote/client.go
+	if r.Method != "POST" {
+		err := fmt.Sprintf("HTTP Method %s instead of POST", r.Method)
+		log.Error("msg", "Write header validation error", "err", err)
+		http.Error(w, err, http.StatusBadRequest)
```

Comment:
Seems like these two things are done a lot of times, maybe we could refactor it into a separate function and cleanup the code.